src/cli/completion.rs

```diff
@@ -215,9 +235,15 @@ _arguments "${_arguments_options[@]}" \
             ;;
         pixi__run)
             opts="-v -q -h --manifest-path --locked --frozen --verbose --quiet --color --help [TASK]..."
-            if [[ ${cur} == -* || ${COMP_CWORD} -eq 2 ]] ; then
+            if [[ ${cur} == -* ]] ; then
                 COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                 return 0
+            elif [[ ${COMP_CWORD} -eq 2 ]]; then
+               local tasks=$(pixi task list --machine-readable 2> /dev/null)
+               if [[ $? -eq 0 ]]; then
+                   COMPREPLY=( $(compgen -W "${tasks}" -- "${cur}") )
+                   return 0
+               fi
```

Comment:
Why is this change needed?

Reply:
Had to dig up the comment from an old squash commit internally. This just changes this test to better match the actual output of clap now. The test is a bit dated for an older version of clap. Since clap is injecting in `pixi task list` and we are replacing that with a insta filter substitution we can test that works correctly.